src/Symfony/Component/Form/Extension/Core/Type/CollectionType.php

```diff
@@ -74,23 +74,45 @@ public function finishView(FormView $view, FormInterface $form, array $options)
      */
     public function configureOptions(OptionsResolver $resolver)
     {
-        $optionsNormalizer = function (Options $options, $value) {
+        $entryOptionsNormalizer = function (Options $options, $value) {
             $value['block_name'] = 'entry';
 
             return $value;
         };
+        $entryType = function (Options $options) {
+            if (null !== $options['type']) {
+                trigger_error('The form option "type" is deprecated since version 2.7 and will be removed in 3.0. Use "entry_type" instead.', E_USER_DEPRECATED);
+
+                return $options['type'];
+            }
+
+            return 'text';
+        };
+        $entryOptions = function (Options $options) {
+            if (null !== $options['options']) {
+                trigger_error('The form option "options" is deprecated since version 2.7 and will be removed in 3.0. Use "entry_options" instead.', E_USER_DEPRECATED);
```

Comment:
"options" will never be `null` if you keep its normalizer. You should move the `trigger_error()` statement to the normalizer of the "options" property instead.